### PR TITLE
Add example script to run NeMo 2.0 llama pretraining with NeMo-Run

### PR DESCRIPTION
**Pull request by hemildesai**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/llm/run/llama_pretraining.py
Comment (ashors1) [inline examples/llm/run/llama_pretraining.py:33]: This description needs to be updated

File: examples/llm/run/llama_pretraining.py
Comment (ShriyaPalsamudram) [inline examples/llm/run/llama_pretraining.py:1]: Change filename to llama3_pretraining?

**Review by ShriyaPalsamudram — DISMISSED** [review]

File: examples/llm/run/llama_pretraining.py
Comment (ShriyaPalsamudram) [inline examples/llm/run/llama_pretraining.py:2]: Add 1-2 lines explaining what this example contains and the pre-reqs. 
1. NeMo-Run from https://github.com/NVIDIA/NeMo-Run
2. Optional slurm, otherwise can be run locally

**Comment by marcromeyn** [comment]
I feel like we should hold off on this until we merge the new entrypoint-API in NeMo-Run, see [docs](https://github.com/NVIDIA/NeMo-Run/blob/c284dcc7005ac5a2183dac661173eee46e05df87/examples/entrypoint/README.md). This will simplify this example considerably while adding more flexibility.

**Comment by ShriyaPalsamudram** [comment]
> I feel like we should hold off on this until we merge the new entrypoint-API in NeMo-Run, see [docs](https://github.com/NVIDIA/NeMo-Run/blob/c284dcc7005ac5a2183dac661173eee46e05df87/examples/entrypoint/README.md). This will simplify this example considerably while adding more flexibility.

But what if we don't want the extra flexibility. The aim of this example is to be able to reproduce exactly what we have been doing. Users can then use CLI or the script itself for personalization.

**Comment by hemildesai** [comment]
Yeah I think we can create a separate CLI folder to host the CLI examples, but would be nice to have both so the users get a clear picture. It will also help show the benefits via a 1:1 comparison.

File: examples/llm/run/llama_pretraining.py
Comment (hemildesai) [reply examples/llm/run/llama_pretraining.py]: Fixed.

**Review by maanug-nv — APPROVED** [review]
no additional concerns

**Comment by marcromeyn** [comment]
@ShriyaPalsamudram if the script would only show how to do one thing, so let's say llama3-8b I would agree. But this script offers some form of parameterization using argparse + the `MODEL_SIZE_MAPPING` dict . I believe when we showcase scripts with a form of parameterization we should be showcasing the functionality inside NeMo-Run that is designed for that.

File: examples/llm/run/llama_pretraining.py
Comment (gautham-kollu) [inline examples/llm/run/llama_pretraining.py:143]: np : Can we add a synthetic dataset/dataloader for this ? so users can run it out-of-the-box

File: examples/llm/run/llama_pretraining.py
Comment (gautham-kollu) [inline examples/llm/run/llama_pretraining.py:35]: np: should we make local = true as deafult because that is the simpler setup.

File: examples/llm/run/llama_pretraining.py
Comment (hemildesai) [reply examples/llm/run/llama_pretraining.py:143]: It comes with the MockDataModule so it should work out of the box.

**Review by ShriyaPalsamudram — APPROVED** [review]

**Pull request merged**
